presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java

```diff
@@ -741,78 +738,69 @@ public static Slice roundLong(@LiteralParameter("s") Long numScale, @SqlType("de
         }
     }
 
-    private static SqlScalarFunction decimalRoundNFunction()
-    {
-        Signature signature = Signature.builder()
-                .kind(SCALAR)
-                .name("round")
-                .longVariableConstraints(
-                        //result precision = increment the input precision only if the input number has a decimal point (scale > 0)
-                        longVariableExpression("return_precision", "min(38, num_precision + 1)"))
-                .argumentTypes(parseTypeSignature("decimal(num_precision,num_scale)", ImmutableSet.of("num_precision", "num_scale")), parseTypeSignature(BIGINT))
-                .returnType(parseTypeSignature("decimal(return_precision,num_scale)", ImmutableSet.of("return_precision", "num_scale")))
-                .build();
-        return SqlScalarFunction.builder(MathFunctions.class)
-                .signature(signature)
-                .description("round to given number of decimal places")
-                .implementation(b -> b
-                        .methods("roundNShortShort", "roundNShortLong", "roundNLongLong")
-                        .withExtraParameters(MathFunctions::decimalRoundNExtraParameters)
-                )
-                .build();
-    }
-
-    private static List<Object> decimalRoundNExtraParameters(SpecializeContext context)
-    {
-        return ImmutableList.of(context.getLiteral("num_precision"), context.getLiteral("num_scale"));
-    }
+    @ScalarFunction("round")
+    @Description("round to nearest integer")
+    public static final class RoundN
+    {
+        @LiteralParameters({"p", "s", "rp"})
+        @SqlType("decimal(rp, s)")
+        @Constraint(variable = "rp", expression = "min(38, p + 1)")
+        public static long roundNShort(
+                @LiteralParameter("p") Long numPrecision,
+                @LiteralParameter("s") Long numScale,
+                @SqlType("decimal(p, s)") long num,
+                @SqlType(BIGINT) long decimals)
```

Comment:
`decimals`? Please rename to something meaningful like `decimalPlaces`. `decimals` is very missleading.